src/libpmem/Makefile

```diff
@@ -48,8 +48,47 @@ SOURCE =\
 	libpmem.c\
 	cpu.c\
 	pmem.c\
-	pmem_linux.c
+	pmem_linux.c\
+	memcpy_nt_avx.c\
+	memcpy_nt_sse2.c\
+	memset_nt_avx.c\
+	memset_nt_sse2.c\
+	memcpy_t_avx.c\
+	memcpy_t_sse2.c\
+	memset_t_avx.c\
+	memset_t_sse2.c
+
+AVX512F_PROG="\#include <immintrin.h>\n\#include <stdint.h>\nint main(){ uint64_t v[8]; __m512i zmm0 = _mm512_loadu_si512((__m512i *)&v); return 0;}"
+AVX512F_AVAILABLE := $(shell printf $(AVX512F_PROG) |\
```

Comment:
$(call check_flag, -mavx512f)

Reply:
Unfortunately this is not enough. With older compilers -mavx512f works, but the first time we use anything from immintrin.h they fail.